default/AI/ShipDesignAI.py

```diff
@@ -1924,6 +1910,77 @@ def _starting_guess(self, available_parts, num_slots):
             ret_val[-1] = num_slots - 1
         return ret_val
 
+
+def _add_shipdesign_fo(design_name, hull, parts_list, model="fighter", description="", icon="", name_desc_in_string_table=False, verbose=False):
+    """This is basically a wrapper around fo.issueCreateShipDesignOrder to also update the cache
+
+    :param design_name: the name of the design
+    :type design_name: str
+    :param description: a human readable description of the design
+    :type description: str
+    :param hull: the name of the hull to use
```

Comment:
> It is indeed the hull name as string.

Well, if you want to be very strict, it's actually a key string, which gets looked up in the stringtables to retrieve the actual _name_ of the hull (that is presented to the player). I guess @Cjkjvfnby was thinking along these lines when he wrote his comment (the same probably applies for similar comments below).

However, AFAIK, on the code level we've always been referring to these key strings as names, contrary to the integer IDs of universe objects, so I think there's no reason not to stick with that convention here.